### PR TITLE
Fix Base function overloading

### PR DESCRIPTION
**Pull request by ararslan**
This gets the MP3 and SampledSignals tests passing for me on master. cc @tkelman

### CODE REVIEW

**Comment by ararslan** [comment]
Hmmmmmmmm uh oh. Looks like something might be wrong with FFTW on nightly now that the binary no longer has FFTW.

**Comment by ararslan** [comment]
Ah, the tests are good on master but not on 0.6. Okay, this isn't the answer then.

**Comment by tkelman** [comment]
Bump? Should we call in @ssfrr or anyone else for some help here? Add upper bounds for the time being?

**Comment by ararslan** [comment]
Haven't been around today. Seems that BinDeps isn't installing FFTW properly so it isn't working on nightly. Trying to figure it out now.

File: src/Filters/Filters.jl [hunk 1]
```diff
@@ -1,8 +1,9 @@
 module Filters
-import ..DSP: @importffts, filt, filt!
+import ..DSP: @importffts
 using Polynomials, ..Util
 import Base: *
 @importffts
+VERSION >= v"0.7.0-DEV.602" && import ..DSP: filt, filt!
```
Comment (ssfrr) [inline src/Filters/Filters.jl]: On 0.6 we need to import these functions from `Base`. Changing this line to 

```julia
if VERSION >= v"0.7.0-DEV.602"
    import ..DSP: filt, filt!
else
    import Base: filt, filt!
end
```

seems to work, but I confess I don't have a full grasp of your FFTW/DSP migration master plan.

Comment (ararslan) [inline src/Filters/Filters.jl]: Thanks, this is a case I forgot to update!

File: src/DSP.jl [hunk 2]
```diff
@@ -24,6 +21,14 @@ end
...
 
+if VERSION >= v"0.7.0-DEV.602"
+    if isdefined(Base, :DSP)
+        import Base: conv, conv2, deconv, filt, filt!, xcorr
+    else
+        export conv, conv2, deconv, filt, filt!, xcorr
+    end
+end
+
 include("dspbase.jl")
```
Comment (ssfrr) [inline src/DSP.jl]: The latest commit fixes things on SampledSignals and MP3, but it's failing its own tests. Looks like `dspbase.jl` also defines `filt` and `filt!` methods. Should that file even be included for Julia 0.6? AFAICT moving the `include("dspbase.jl")` to inside the `if VERSION >= v"0.7.0-DEV.602"` block above fixes things on 0.6 and 0.7 (nightly)

Comment (ssfrr) [inline src/DSP.jl]: @ararslan Do you think this fix will get things working? Or at least not cause extra problems on 0.7 while getting things working on 0.6?

Comment (ararslan) [inline src/DSP.jl]: Sorry for the delay, been pretty absent on GitHub due to JuliaCon. I'll try that fix now.

Comment (ssfrr) [inline src/DSP.jl]: Totally understand. I'm guessing/hoping that there will be folks from my poster session that might want to try out the audio stuff now that our conference is wrapping up.

**Comment by ararslan** [comment]
Nightly failing due to FFTW garbage. Can we just merge the fix we have for that figure out Windows later, @tkelman?

**Comment by tkelman** [comment]
Nightly doesn't really matter yet. Fix 0.6.

**Comment by ararslan** [comment]
0.6 is passing tests here

**Comment by ssfrr** [comment]
did you fix the `dspbase.jl` issue?

**Comment by ararslan** [comment]
Yeah I pushed that and squashed the commits

**Comment by ssfrr** [comment]
Sounds good. Passes the latest tagged versions of SampledSignals and MP3 as well.

**Comment by ararslan** [comment]
Awesome, thanks. I'll merge and tag then.

**Pull request merged**